src/fromager/settings.py

```diff
@@ -143,16 +143,46 @@ def _resolve_template(
         raise
 
 
-def _parse(content: str) -> Settings:
-    data = yaml.safe_load(content)
-    return Settings(data)
+def load(settings_file: pathlib.Path, settings_dir: pathlib.Path) -> Settings:
+    settings_data = {}
 
-
-def load(filename: pathlib.Path) -> Settings:
-    filepath = pathlib.Path(filename)
+    filepath = pathlib.Path(settings_file)
     if not filepath.exists():
         logger.debug("settings file %s does not exist, ignoring", filepath.absolute())
-        return Settings({})
-    with open(filepath, "r") as f:
-        logger.info("loading settings from %s", filepath.absolute())
-        return _parse(f.read())
+    else:
+        with open(filepath, "r") as f:
+            logger.info("loading settings from %s", filepath.absolute())
+            settings_data = yaml.safe_load(f.read())
+
+    # Per-package files are inserted in the `packages` key using the name that
+    # will be used to look the value up. Transform any existing keys to that
+    # format so we can warn if there are overriding values.
+    package_settings_from = {}
+    if "packages" not in settings_data:
+        settings_data["packages"] = {}
+    else:
+        new_packages = {}
+        for name, value in settings_data["packages"].items():
+            package_name = overrides.pkgname_to_override_module(name)
+            new_packages[package_name] = value
```

Comment:
I think the `new_packages` variable is not being used anymore?

Reply:
Ooo, that's a bug. I updated the test in the new version of the PR and made sure to replace the updated packages section.